src/event.js

```diff
@@ -25,14 +25,35 @@ const enrichMeta = (event, appName, ipv4) => {
   return enrichedEvent;
 };
 
-module.exports = (kinesis, event, config) => {
+const emitEvent = (kinesis, event, config) => {
   const enrichedEvent = enrichMeta(event, config.appName, config.ipv4);
 
   const params = {
     Data: JSON.stringify(enrichedEvent),
     PartitionKey: config.partitionKey || partitionKey(enrichedEvent),
     StreamName: config.kinesisStream.resource
   };
-
   return kinesis.putRecord(params).promise();
 };
+
+const emitEvents = (kinesis, events, config) => {
+  const records = events.map(event => {
+    const enrichedEvent = enrichMeta(event, config.appName, config.ipv4);
+    return {
+      Data: JSON.stringify(enrichedEvent),
+      PartitionKey: config.partitionKey || partitionKey(enrichedEvent),
+    }
+  })
+
+  const params = {
+    Records: records,
+    StreamName: config.kinesisStream.resource
+  };
+
+  return kinesis.putRecords(params).promise();
```

Comment:
I think retry mechanism should be left to consumers honestly.
Each project's retry needs might be very different from the next one, and thus it should be easy enough for consumers to find something that fits their needs.

Reply:
I don't think so, mainly because it's tightly coupled to AWS API, in the end that's also the whole purpose of this pkg - to abstract those details away ;-)

Reply:
This package already supports retires which can be configured [here](https://github.com/babbel/miza-kinesis/blob/master/src/kinesis.js#L6), this will automatically retry requests like 5xx or throttling errors, but the client errors ([4xx](https://docs.aws.amazon.com/kinesis/latest/APIReference/CommonErrors.html)) indicate that you need to revise the request to correct the problem before trying again. so, I would leave it to consumers to take any action at theirs end.


If we still need to discuss this, I think it's best we scope out retrying logic to the next version. The current package that supports only emitting of one event also doesn't support retries, maybe we could discuss this topic with the team that owns and see their reason for not implementing it in the first place.

Reply:
One level of abstraction that can be thought of is pulling out the retry mechanism from consumers and creating a framework that provides an interface to the consumers. Perhaps something to be considered for future scope.

Reply:
I would maybe consider putting some default retry mechanism here, and provide a user with an option/interface to update it, I am not sure how much effort it might be to implement, and maybe if it can just be an additional ticket?

Reply:
yeah, let's discuss this is another ticket.